YotiButtonSDK/Kernel/KernelSDK.swift

```diff
@@ -55,21 +49,21 @@ class KernelSDK: NSObject {
                                          URLQueryItem(name: "sourceScheme", value: sourceSchemes.first)]
 
             guard let url = urlComponents?.url else {
-                delegate.yotiSDKDidFail(for: useCaseID, with: GenericError.malformedValue("qrCodeURL"))
+                delegate.yotiSDKDidFail(for: useCaseID, with: ShareRequestError.startScenarioError("Malformed value received"))
                 return
             }
 
             guard UIApplication.shared.canOpenURL(url) else {
-                delegate.yotiSDKDidFail(for: useCaseID, with: GenericError.unknown(url.absoluteString))
+                delegate.yotiSDKDidFail(for: useCaseID, with: SetupError.invalidApplicationQueriesSchemes(url))
 
                 print("Cannot Open Yoti, please check LSApplicationQueriesSchemes or install Yoti")
                 return
             }
```

Comment:
Not sure this check is quite valid; in practice to check the `LSApplicationQueriesSchemes` we can read it from the bundle as above with the URL schemes: `Bundle.main.object(forInfoDictionaryKey: "LSApplicationQueriesSchemes")`. This only checks we can open a URL that doesn't follow the format associated with the print statement and thus error.